### PR TITLE
doc: Update examples for Immer and Valtio

### PR DESCRIPTION
**Pull request by dmaskasky**
## Related Issues or Discussions
Resolves: https://github.com/pmndrs/jotai/discussions/2171

## Summary

Update the inline and codesandbox examples for Immer and Valtio to more accurately and clearly showcase their functionalities.

## Check List

- [x] `yarn run prettier` for formatting code and docs

### CODE REVIEW

File: docs/integrations/immer.mdx [hunk 1]
```diff
@@ -24,17 +24,20 @@ The signature of writeFunction is `(get, set, update: (draft: Draft<Value>) => v
 import { useAtom } from 'jotai'
 import { atomWithImmer } from 'jotai-immer'
 
-const countAtom = atomWithImmer(0)
+const countAtom = atomWithImmer({ value: 0 })
```
Comment (dai-shi) [inline docs/integrations/immer.mdx:27]: This doesn't show what we would like to show. Let's keep `atomWithImmer(0)`.

File: docs/integrations/immer.mdx [hunk 2]
```diff
@@ -24,17 +24,20 @@ The signature of writeFunction is `(get, set, update: (draft: Draft<Value>) => v
...
 const Counter = () => {
   const [count] = useAtom(countAtom)
-  return <div>count: {count}</div>
+  return <div>count: {count.value}</div>
 }
 
 const Controls = () => {
   const [, setCount] = useAtom(countAtom)
   // setCount === update : (draft: Draft<Value>) => void
-  const inc = () => setCount((c) => (c = c + 1))
```
Comment (dai-shi) [inline docs/integrations/immer.mdx:37]: How about `(draft) => ++draft`?

File: docs/integrations/immer.mdx [hunk 3]
```diff
@@ -53,18 +56,21 @@ Check this example with atomWithImmer:
 import { useAtom, atom } from 'jotai'
 import { withImmer } from 'jotai-immer'
 
-const primitiveAtom = atom(0)
+const primitiveAtom = atom({ value: 0 })
```
Comment (dai-shi) [inline docs/integrations/immer.mdx:59]: Same for this.
```suggestion
const primitiveAtom = atom(0)
```

File: docs/integrations/immer.mdx [hunk 4]
```diff
@@ -83,17 +89,20 @@ This hook takes an atom and replaces the atom's `writeFunction` with the new imm
 import { useAtom } from 'jotai'
 import { useImmerAtom } from 'jotai-immer'
 
-const primitiveAtom = atom(0)
+const primitiveAtom = atom({ value: 0 })
```
Comment (dai-shi) [inline docs/integrations/immer.mdx:92]:
```suggestion
const primitiveAtom = atom(0)
```

File: docs/integrations/valtio.mdx [hunk 5]
```diff
@@ -32,21 +32,29 @@ It's two-way binding and you can change the value from both ends.
 ```jsx
 import { useAtom } from 'jotai'
 import { atomWithProxy } from 'jotai-valtio'
-import { proxy } from 'valtio/vanilla'
+import { proxy, useSnapshot } from 'valtio'
```
Comment (dai-shi) [inline docs/integrations/valtio.mdx:35]: I don't think we need to show the use of `useSnapshot` as a basic example.

File: docs/integrations/valtio.mdx [hunk 6]
```diff
@@ -32,21 +32,29 @@ It's two-way binding and you can change the value from both ends.
...
         }>
-        button
+        inc atom
+      </button>
+      <button
+        onClick={() => {
+          ++proxyState.count
+        }}>
+        inc proxy
       </button>
```
Comment (dai-shi) [inline docs/integrations/valtio.mdx:58]: Adding this looks nice. Thanks.

File: docs/integrations/valtio.mdx [hunk 7]
```diff
@@ -82,7 +90,7 @@ const countProxyAtom = mutableAtom(0)
 
 function IncrementButton() {
   const countProxy = useAtomValue(countProxyAtom)
-  return <button onClick={() => countProxy.value++}>+</button>
+  return <button onClick={() => ++countProxy.value}>+</button>
```
Comment (dai-shi) [inline docs/integrations/valtio.mdx:93]: Yeah, this and the following changes make the style consistent.

**Review by dai-shi — CHANGES_REQUESTED** [review]
Thanks for working on the improvements. Some suggestions.

File: docs/integrations/immer.mdx (hunk 1 above)
Comment (dmaskasky) [reply docs/integrations/immer.mdx:27]: Doesn't work: https://codesandbox.io/s/crazy-goldstine-jl5vdm

File: docs/integrations/immer.mdx (hunk 2 above)
Comment (dmaskasky) [reply docs/integrations/immer.mdx:37]: This doesn't work for me: https://codesandbox.io/s/crazy-goldstine-jl5vdm

Comment (dai-shi) [reply docs/integrations/immer.mdx:37]: Oh, you are absolutely right! How come this has been left for such a long time.

File: docs/integrations/immer.mdx (hunk 1 above)
Comment (dai-shi) [reply docs/integrations/immer.mdx:27]: my bad

File: docs/integrations/immer.mdx (hunk 4 above)
Comment (dai-shi) [reply docs/integrations/immer.mdx:92]: my bad

File: docs/integrations/immer.mdx (hunk 3 above)
Comment (dai-shi) [reply docs/integrations/immer.mdx:59]: my bad

File: docs/integrations/immer.mdx (hunk 1 above)
Comment (dmaskasky) [reply docs/integrations/immer.mdx:27]: I don't think `0` adequately demonstrates immer. Immer works well with objects but not with primitives. For example, `++draft` does not work as expected.

File: docs/integrations/immer.mdx (hunk 3 above)
Comment (dmaskasky) [reply docs/integrations/immer.mdx:59]: same issue

File: docs/integrations/immer.mdx (hunk 4 above)
Comment (dmaskasky) [reply docs/integrations/immer.mdx:92]: same concern

File: docs/integrations/valtio.mdx [hunk 8]
```diff
@@ -32,7 +32,7 @@ It's two-way binding and you can change the value from both ends.
 ```jsx
 import { useAtom } from 'jotai'
 import { atomWithProxy } from 'jotai-valtio'
-import { proxy } from 'valtio/vanilla'
+import { proxy } from 'valtio/vanilla'
```
Comment (dai-shi) [inline docs/integrations/valtio.mdx:35]:
```suggestion
import { proxy } from 'valtio/vanilla'
```

File: docs/integrations/valtio.mdx [hunk 9]
```diff
@@ -42,11 +42,18 @@ const Counter = () => {
   return (
     <>
       count: {state.count}
+      proxyState: {proxyState.count}
```
Comment (dai-shi) [inline docs/integrations/valtio.mdx:45]: This is confusing, as it's not reactive and usually bad practice in Valtio.
```suggestion
```

**Review by dai-shi — APPROVED** [review]
Sorry for my misunderstanding. This should be good. Thanks for your contribution.

**Pull request merged**